### PR TITLE
Getting rid of float in simulation code Part 2

### PR DESCRIPTION
**Pull request by reaperrr**
Covers the rest of examples listed in #10845 + `ThrowsParticle`.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/ThrowsParticle.cs [hunk 1]
```diff
@@ -53,8 +53,8 @@ class ThrowsParticle : ITick
 		int tick = 0;
 		int length;
 
-		float facing;
-		float rotation;
+		int facing;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/ThrowsParticle.cs]: Could you make these `WAngle`, and then do `() => facing.Angle / 4` in the Animation ctor?  That makes it clearer what these are for, and has a side-effect of giving you a bit more precision.

**Comment by pchote** [comment]
:+1: after that.

**Comment by reaperrr** [comment]
@pchote Updated, but that required a bit more code changes than I had hoped, and I'm not confident in the code (though it seems to work fine in-game), so I kept it as separate commit for now.

File: OpenRA.Mods.Common/Traits/ThrowsParticle.cs [hunk 2]
```diff
@@ -64,7 +64,7 @@ public ThrowsParticle(ActorInitializer init, ThrowsParticleInfo info)
 
 			// TODO: Carry orientation over from the parent instead of just facing
 			var bodyFacing = init.Contains<FacingInit>() ? init.Get<FacingInit, int>() : 0;
-			facing = Turreted.GetInitialTurretFacing(init, 0);
+			facing = new WAngle(Turreted.GetInitialTurretFacing(init, 0));
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/ThrowsParticle.cs]: Use WAngle.FromFacing here.

File: OpenRA.Mods.Common/Traits/ThrowsParticle.cs [hunk 3]
```diff
@@ -76,9 +76,9 @@ public ThrowsParticle(ActorInitializer init, ThrowsParticleInfo info)
 			length = (finalPos - initialPos).Length / info.Velocity;
 
 			// Facing rotation
-			rotation = WDist.FromPDF(Game.CosmeticRandom, 2).Length * info.TurnSpeed / 1024;
+			rotation = new WAngle(WDist.FromPDF(Game.CosmeticRandom, 2).Length * info.TurnSpeed / 1024);
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/ThrowsParticle.cs]: FromFacing

**Comment by reaperrr** [comment]
Updated.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/3042ea078f707eba41d50645cf4bf6790e931634...1859227b12a757093693056d5012051c1950e8eb)

**Pull request merged**